src/client/Statistics.ts

```diff
@@ -20,7 +20,50 @@ import {
  * first would be .2*2=.4
  */
 export function calculateAverages(responseData: GetFullScoresResponseData): SubmissionAverageStatistics {
-  throw new Error("Not Yet Implemented");
+  let cloned = [...responseData.assignments];
+  let subIndices: Array<number | undefined> = [];
+
+  //The for loop iterates through the assignments array
+  //Then it checks if the type is 1 and it is not active
+  //Then if subIndices array doesnt contain the respective submission Index, then add it to subIndices array
+  for (var i = 0; i < cloned.length; i++) {
+    if (cloned[i].type === 1 && !(cloned[i].active)) {
+      if ((!subIndices.includes(cloned[i].submissionIndex)) && (!cloned[i].submissionIndex === undefined)) {
+        subIndices.push(cloned[i].submissionIndex);
+      }
+    }
+  }
+  //After for loop, subIndices array is populated with unique submission Inds
+
+  //Now, run through subInd array and populate result array with scores for each submission from cloned
+  let avgArray: Array<SubmissionAverageStatistics> = [];;
+  for (var j = 0; j < subIndices.length; j++) {
+    var average = 0;
+    var count = 0;
+    var sum = 0;
+    for (var k = 0; k < cloned.length; k++) {
+      if (cloned[k].submissionIndex === subIndices[j]) {
+        if (cloned[k].rating === undefined) {
+          sum = 0;
+          count = 0;
+          continue;
+        }
+        else {
+          sum = cloned[k].rating + sum;
+          count++;
+        }
+      }
+      else {
+        sum = 0;
+        count = 0;
+        continue;
+      }
+    }
+    average = sum/count;
+    avgArray.push(subIndices[j],average,0);
+  }
+
+  return avgArray;
```

Comment:
You should return an array of objects, for instance `[{id: 1, plainAverage: .5}, {id: 2, plainAverage: 3}]` could be a valid return.

Reply:
You can also remove `judgeWeightedAverage` from the interface. That's lower priority and can be done in a different PR.